src/main.rs

```diff
@@ -791,8 +791,10 @@ fn object_files(cx: &Context) -> Result<Vec<OsString>> {
     // This is not the ideal way, but the way unstable book says it is cannot support them.
     // https://doc.rust-lang.org/nightly/rustc/instrument-coverage.html#tips-for-listing-the-binaries-automatically
     let mut target_dir = cx.ws.target_dir.clone();
+    let mut build_dir = cx.ws.metadata.build_directory().to_owned();
```

Comment:
This is likely inefficient as it triggers a rebuild, but unfortunately, build-dir lacks a flag like --target-dir and only has transparent environment variables, so it may be difficult to make it behave the same as target-dir here.